modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/model/listener/test/LayoutModelListenerTest.java

```diff
@@ -0,0 +1,190 @@
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group.getCompanyId(), _group.getGroupId(),
+			TestPropsValues.getUserId());
+	}
+
+	@Test
+	public void testOnBeforeUpdate() throws Exception {
+		Layout layout1 = LayoutTestUtil.addTypeContentLayout(_group);
+
+		long defaultSegmentsExperienceId =
+			_segmentsExperienceLocalService.fetchDefaultSegmentsExperienceId(
+				layout1.getPlid());
+
+		SegmentsExperiment segmentsExperiment =
+			_segmentsExperimentLocalService.addSegmentsExperiment(
+				defaultSegmentsExperienceId,
+				_portal.getClassNameId(Layout.class), layout1.getPlid(),
+				"AB test", "A/B test description",
+				SegmentsExperimentConstants.Goal.BOUNCE_RATE.getLabel(),
+				StringPool.BLANK, _serviceContext);
+
+		Object asahSegmentsExperimentProcessor =
+			ReflectionTestUtil.getFieldValue(
+				_modelListener, "_asahSegmentsExperimentProcessor");
+
+		Object asahFaroBackendClient = ReflectionTestUtil.getFieldValue(
+			asahSegmentsExperimentProcessor, "_asahFaroBackendClient");
+
+		ReflectionTestUtil.setFieldValue(
+			asahFaroBackendClient, "_http",
+			MockHttpUtil.geHttp(
+				Collections.singletonMap(
+					"/api/1.0/experiments/" +
+						segmentsExperiment.getSegmentsExperienceId(),
+					() -> "{}")));
+
+		SegmentsExperience segmentsExperience =
+			_segmentsExperienceLocalService.addSegmentsExperience(
+				TestPropsValues.getUserId(), _group.getGroupId(), 0,
+				_portal.getClassNameId(Layout.class), layout1.getPlid(),
+				Collections.singletonMap(
+					LocaleUtil.getSiteDefault(), "Variant 1"),
+				false, new UnicodeProperties(true), _serviceContext);
+
+		_segmentsExperimentRelService.addSegmentsExperimentRel(
+			segmentsExperiment.getSegmentsExperimentId(),
+			segmentsExperience.getSegmentsExperienceId(), _serviceContext);
+
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId",
+							RandomTestUtil.randomLong()
+						).put(
+							"liferayAnalyticsEnableAllGroupIds", true
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							RandomTestUtil.randomString()
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			String title = "Layout title modified";
+
+			layout1.setTitleMap(
+				Collections.singletonMap(LocaleUtil.getSiteDefault(), title));
+
+			_layoutLocalService.updateLayout(layout1);
+
+			Layout layout2 = _layoutLocalService.getLayout(layout1.getPlid());
+
+			Assert.assertEquals(
```

Comment:
You are asserting the update, but someone has should tested this method, what you should assert is that the AC endpoint is correctly called.

And right now this is not possible, because we are calling _asahFaroBackendClient.updateExperiment and this doesn't have any changes that we can assert, so this method can't be tested with an Integration Tests.

What do you think to test it using unit tests? It is possible? does it give you any value? Maybe we should not test it at all.

Reply:
This class has been deleted and I added some unit tests of AsahFaroBackendClientImplTest class.